src/Console/Commands/EmitEvents.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Softonic\TransactionalEventPublisher\Console\Commands;
+
+use Exception;
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\LazyCollection;
+use RuntimeException;
+use Softonic\TransactionalEventPublisher\Contracts\EventStoreMiddlewareContract;
+use Softonic\TransactionalEventPublisher\Model\DomainEvent;
+use Softonic\TransactionalEventPublisher\Model\DomainEventsCursor;
+
+class EmitEvents extends Command
+{
+    /**
+     * Max delay for errors retry in seconds (1 minute).
+     */
+    private const MAX_DELAY_FOR_ERRORS = 60;
+
+    /**
+     * Max delay for no events retry in microseconds (1 second).
+     */
+    private const MAX_DELAY_FOR_NO_EVENTS = 1_000_000;
+
+    /**
+     * Base delay for no events retry in microseconds (1 millisecond).
+     */
+    private const BASE_DELAY_FOR_NO_EVENTS = 1000;
+
+    protected $signature = 'event-sourcing:emit
+        {--dbConnection=mysql : Indicate the database connection to use (MySQL unbuffered for better performance when large amount of events)}
+        {--batchSize=100 : Indicate the amount of events to be sent per publish. Increase for higher throughput}
+        {--allEvents : Option to send all the events from the beginning by resetting the cursor}';
+
+    protected $description = 'Continuously emits domain events in batches';
+
+    public EventStoreMiddlewareContract $eventPublisherMiddleware;
+
+    public DomainEventsCursor $cursor;
+
+    public string $databaseConnection;
+
+    public int $batchSize;
+
+    public int $attemptForErrors = 1;
+
+    public int $attemptForNoEvents = 1;
+
+    private bool $eventsProcessed;
+
+    public function handle(EventStoreMiddlewareContract $eventPublisherMiddleware): void
+    {
+        $this->eventPublisherMiddleware = $eventPublisherMiddleware;
+
+        $this->databaseConnection = $this->option('dbConnection');
+        $this->batchSize = (int)$this->option('batchSize');
+        $resetCursor = $this->option('allEvents');
+
+        $this->cursor = $this->getInitialCursor($resetCursor);
+
+        $this->sendBatches();
+    }
+
+    private function getInitialCursor(bool $resetCursor): DomainEventsCursor
+    {
+        $cursor = DomainEventsCursor::first();
+
+        if (empty($cursor)) {
+            $cursor = new DomainEventsCursor(['last_id' => 0]);
+            $cursor->save();
+
+            return $cursor;
+        }
+
+        if ($resetCursor) {
+            $cursor->update(['last_id' => 0]);
+        }
+
+        return $cursor;
```

Comment:
We don't need to check the cursor if we are reciving a $resetCursor
```suggestion
        if (!$resetCursor) {
            $cursor = DomainEventsCursor::first();
        }

        if (empty($cursor)) {
            $cursor = new DomainEventsCursor(['last_id' => 0]);
            $cursor->save();

            return $cursor;
        }

        return $cursor;
```

Reply:
Your logic is not resetting the cursor, in case it exists, updating it to 0.